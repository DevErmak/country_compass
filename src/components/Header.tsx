import { useSelector } from 'react-redux';
import { ReactComponent as Logo } from '../static/images/logo.svg';
// import AsyncSelect from 'react-select/async';
import { getFullInfoCountry, getOptionsCountry } from '../store/country/fullInfoCountrySelectors';
import Select from 'react-select';

type Props = {};

export default function Header({}: Props) {
  const fullInfoCountry = useSelector(getOptionsCountry);

  // const getDataCity = () => {
  //   return fullInfoCountry.map((item) => ({
  //     value: item.name.common,
  //     label: item.name.common,
  //   }));
  // };

  // const FindDataCity = (inputValue:string) => {
  //   return fullInfoCountry.filter((item) => { if (inputValue === item.name.common) {
  //     value: item.name.common,
  //     label: item.name.common,
  //   });
  // };

  // let timerId: ReturnType<typeof setTimeout>;

  // const promiseOptions = (inputValue: string) => {
  //   return new Promise((resolve, reject) => {
  //     clearTimeout(timerId);
  //     timerId = setTimeout(async () => {
  //       try {
  //         const results = await getDataCity;
  //         resolve(results);
  //       } catch (error) {
  //         reject(error);
  //       }
  //     }, 1000);
  //   });
  // };
  {
    /* <AsyncSelect
          classNamePrefix="custom-select-weather"
          onChange={handleChange}
          components={{ DropdownIndicator: customIcon }}
          placeholder="Введите название страны"
          loadOptions={promiseOptions}
          value={nameCity}
        /> */
  }
  return (
    <div className="header-container">
      <div className="find-country">
        <Select options={fullInfoCountry} />
      </div>
      <Logo width={90} height={80} />
    </div>
  );
}
