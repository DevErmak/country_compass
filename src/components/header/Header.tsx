import { Link, redirect, useNavigate } from 'react-router-dom';
import './header.css';
import { useDispatch } from 'react-redux';
import { getFullInfoCountryClear } from '../../store/country/infoCountrySlice';
import { useSelector } from 'react-redux';
import {
  getIsActiveModal,
  getIsAuthentication,
  getListFavoriteCountries,
  getState,
  getUserName,
} from '../../store/user/userSelectors';
import { setActiveModal, setAuthentication } from '../../store/user/infoUserSlice';
import Modal from '../modal/Modal';
import ReactDOM from 'react-dom';
import { useState } from 'react';
import FormLogin from '../login/FormLogin';
import FormRegister from '../register/FormRegister';

type Props = {};

export default function Header({}: Props) {
  const navigate = useNavigate();
  const listFavorite = useSelector(getListFavoriteCountries);

  const isLogin = useSelector(getIsAuthentication);

  const dispatch = useDispatch();
  const isActiveModal = useSelector(getIsActiveModal);
  console.log('---------------->isActiveModal', isActiveModal);
  const State = useSelector(getState);
  console.log('---------------->State', State);

  const [isRegister, setIsRegister] = useState(false);
  const userName = useSelector(getUserName);

  const ClickOnNameSite = () => {
    dispatch(getFullInfoCountryClear());
  };

  const ClickOnLogOut = () => {
    let userInfo = JSON.parse(localStorage.getItem('userInfo') as string);
    console.log('123213-----------123213----->userInfo', userInfo);
    console.log('123213-----------123213----->userName', userName);

    if (userInfo !== null) {
      const user = userInfo.find((item: any) => item.name === userName);
      if (user !== undefined) {
        console.log('123123---------------->user', user);
        const newUserInfo = userInfo.filter((item: any) => user !== item);
        user.listFavorite = listFavorite;
        newUserInfo.push(user);
        localStorage.setItem('userInfo', JSON.stringify(newUserInfo));
      }
    }
    dispatch(setAuthentication(false));
  };

  const ClickOnLogin = () => {
    dispatch(setActiveModal(true));
    setIsRegister(false);
  };

  const ClickOnRegister = () => {
    dispatch(setActiveModal(true));
    setIsRegister(true);
  };

  if (isLogin)
    return (
      <div className="container-header">
        <div className="name-site" onClick={() => ClickOnNameSite()}>
          Europe.know
        </div>
        <div className="spacer"></div>
        <div className="logout" onClick={() => ClickOnLogOut()}>
          Logout
        </div>
        <div className="my-countries" onClick={() => navigate('my-countries')}>
          My countries
        </div>
      </div>
    );
  else
    return (
      <>
        <div className="container-header">
          <div className="name-site" onClick={() => ClickOnNameSite()}>
            Europe.know
          </div>
          <div className="spacer"></div>
          <div className="login" onClick={() => ClickOnLogin()}>
            Login
          </div>
          <div className="register" onClick={() => ClickOnRegister()}>
            Register
          </div>
        </div>
        {isRegister ? (
          <Modal>
            <FormRegister />
          </Modal>
        ) : (
          <Modal>
            <FormLogin />
          </Modal>
        )}
      </>
    );
}
