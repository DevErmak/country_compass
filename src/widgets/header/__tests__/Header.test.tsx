import Header from '../ui/no-auth-header';
import { renderComponentWithStore } from '../../../shared/lib/test-helpers';

it('renders correctly Header', () => {
  const { container } = renderComponentWithStore({
    Component: Header,
    preloadedState: {
      infoCountries: {
        listCountries: [],
        fullInfoCountry: [],
        isFullInfoCountry: false,
        isLoading: false,
        infoErrorResponse: '',
      },
      infoUser: {
        userName: '',
        listFavoriteCountries: [],
        isAuthentication: false,
        infoModal: { isActiveModal: false, formModal: 'login' },
      },
    },
  });
  expect(container).toMatchSnapshot();
});
