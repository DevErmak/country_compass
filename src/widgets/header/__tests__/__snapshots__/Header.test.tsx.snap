// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Header 1`] = `
<div>
  <div
    class="container-header"
  >
    <div
      class="name-site"
    >
      Europe.know
    </div>
    <div
      class="spacer"
    />
    <div
      class="login"
    >
      Login
    </div>
    <div
      class="register"
    >
      Register
    </div>
  </div>
  <div
    id="portal"
  />
</div>
`;
